### PR TITLE
AMIGAOS: Preparations for use with AmiUpdate

### PR DESCRIPTION
**Pull request by raziel-**
Adding a bit of code to make ScummVM AmiUpdate aware.
Everything else will be dealt with through a (yet to finish) AmigaOS installer script and the entries in AmiUpdate's database.

I'd like to ask for hints regarding
- Code Formatting
- Optimizations
- Comments
- Obivous errors
or
- Ill-designed code.

It's not much code, but i'm sure i still messed up glorious either way

### CODE REVIEW

File: backends/platform/sdl/amigaos/amigaos-main.cpp [hunk 1]
```diff
@@ -24,13 +24,44 @@
...
-	// Set up a stack cookie to avoid crashes due to too few stack set by users 
+	// The following will gather the application name and add the install path
+	// to a variable in AmigaOS4's ENV(ARC) system. It will be placed in AppPaths
+	// so that ScummVM can become AmiUpdate aware
+	const char *const appname = "ScummVM";
+
+	BPTR lock;
+	APTR oldwin;
+
+		// Obtain the lock to the home directory
```
Comment (criezy) [inline backends/platform/sdl/amigaos/amigaos-main.cpp]: It looks like indentation is wrong starting from this line until line 50, and then wrong again on lines 61 and 62 (one surnumerous tab).

File: backends/platform/sdl/amigaos/amigaos-main.cpp [hunk 2]
```diff
@@ -24,13 +24,44 @@
...
+		if(( lock = IDOS->GetProgramDir() )) {
```
Comment (criezy) [inline backends/platform/sdl/amigaos/amigaos-main.cpp]: Space after if. Also I think that usually we don't use a space after the opening parenthesis and before the closing one.
And why use two sets of parentheses? One would be enough.

File: backends/platform/sdl/amigaos/amigaos-main.cpp [hunk 3]
```diff
@@ -24,13 +24,44 @@
...
+			TEXT progpath[2048];
+			TEXT apppath[1024] = "AppPaths";
+
+			if( IDOS->DevNameFromLock( lock,
```
Comment (criezy) [inline backends/platform/sdl/amigaos/amigaos-main.cpp]: See comment above for if formatting

File: backends/platform/sdl/amigaos/amigaos-main.cpp (hunk 2 above)
Comment (raziel-) [inline backends/platform/sdl/amigaos/amigaos-main.cpp]: If i remove the second set of parentheses i get a compiler warning:

    C++      backends/platform/sdl/amigaos/amigaos-main.o
backends/platform/sdl/amigaos/amigaos-main.cpp: In function 'int main(int, char**)':
backends/platform/sdl/amigaos/amigaos-main.cpp:43:34: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
  if (lock = IDOS->GetProgramDir()) {
                                   (Points to the last parantheses)    ^

**Comment by raziel-** [comment]
@criezy 

Thank you very much for the hints

File: backends/platform/sdl/amigaos/amigaos-main.cpp [hunk 4]
```diff
@@ -24,13 +24,44 @@
...
+
+		// Stop any "Insert volume..." type requesters
+		oldwin = IDOS->SetProcWindow((APTR)-1);
+
+		// Finally, set the variable to the path the executable was run from
+		IDOS->AddPart( apppath, appname, 1024);
+		IDOS->SetVar( apppath, progpath, -1, GVF_GLOBAL_ONLY|GVF_SAVE_VAR );
+
+		// Turn system requesters back on
+		IDOS->SetProcWindow( oldwin );
```
Comment (criezy) [inline backends/platform/sdl/amigaos/amigaos-main.cpp]: Indentation is now wrong on lines 52 to 60 above.

File: backends/platform/sdl/amigaos/amigaos-main.cpp [hunk 5]
```diff
@@ -24,13 +24,44 @@
...
+	if ((lock = IDOS->GetProgramDir())) {
+		TEXT progpath[2048];
+		TEXT apppath[1024] = "AppPaths";
+
+		if (IDOS->DevNameFromLock(lock,
+			progpath,
+			sizeof(progpath),
+			DN_FULLPATH)) {
+
+	// Stop any "Insert volume..." type requesters
```
Comment (sev-) [inline backends/platform/sdl/amigaos/amigaos-main.cpp]: Indentation. You're using spaces instead of tabs.

Comment (raziel-) [inline backends/platform/sdl/amigaos/amigaos-main.cpp]: Where?

I used the online editor to fix the formerly broken identation and it's still not correct? :-(

Comment (sev-) [inline backends/platform/sdl/amigaos/amigaos-main.cpp]: Just take a look at lines 60 and 61. You may clearly see that the whole block 52-60 is badly indented.

Comment (raziel-) [inline backends/platform/sdl/amigaos/amigaos-main.cpp]: There seems to be a problem with my browser displaying those lines.

In all the cases (display and edit) all those lines show that there are tabs, not spaces.
If i place the cursor on those lines and move them right it's clearly a tab

I don't know what to change, i did it with the online editor once and if it's still wrong i'm not sure how to change it :-(

**Comment by sev-** [comment]
OK, I'll do it in your behalf then. Merged.

**Pull request merged**